JUnit version 4.11
....I...........................................................................................................................................................................................................................................................................................................................org.mockito.internal.exceptions.ExceptionIncludingMockitoWarnings: contains both: actual test failure *and* Mockito warnings.
unused stubbing blah
 *** The actual failure is because of: ***

....................................................................................................................................................................................................II.......EI.I........III...............................********************************
*** Mockito interactions log ***
********************************
mock.giveMeSomeString("arg");
 invoked: -> at org.mockitousage.debugging.PrintingInvocationsDetectsUnusedStubTest.shouldDetectUnusedStubbingWhenPrinting(PrintingInvocationsDetectsUnusedStubTest.java:22)
********************************
***       Unused stubs       ***
********************************
mock.giveMeSomeString("different arg");
 stubbed: -> at org.mockitousage.debugging.PrintingInvocationsDetectsUnusedStubTest.shouldDetectUnusedStubbingWhenPrinting(PrintingInvocationsDetectsUnusedStubTest.java:21)

.********************************
*** Mockito interactions log ***
********************************
mock.giveMeSomeString("arg");
 invoked: -> at org.mockitousage.debugging.PrintingInvocationsWhenEverythingOkTest.businessLogicWithAsking(PrintingInvocationsWhenEverythingOkTest.java:34)
 stubbed: -> at org.mockitousage.debugging.PrintingInvocationsWhenEverythingOkTest.performStubbing(PrintingInvocationsWhenEverythingOkTest.java:30)
mockTwo.doSomething("foo");
 invoked: -> at org.mockitousage.debugging.PrintingInvocationsWhenEverythingOkTest.businessLogicWithTelling(PrintingInvocationsWhenEverythingOkTest.java:39)

.********************************
*** Mockito interactions log ***
********************************
foo.giveMeSomeString("arg");
 invoked: -> at org.mockitousage.debugging.PrintingInvocationsWhenStubNotUsedTest.businessLogicWithAsking(PrintingInvocationsWhenStubNotUsedTest.java:34)
foo.doSomething(null);
 invoked: -> at org.mockitousage.debugging.PrintingInvocationsWhenStubNotUsedTest.businessLogicWithTelling(PrintingInvocationsWhenStubNotUsedTest.java:39)
********************************
***       Unused stubs       ***
********************************
foo.giveMeSomeString("different arg");
 stubbed: -> at org.mockitousage.debugging.PrintingInvocationsWhenStubNotUsedTest.performStubbing(PrintingInvocationsWhenStubNotUsedTest.java:30)

..............II..........................I......................................................................................I...............................I................................................................................................................................................................................................................................................................................................I...
Time: 3.322
There was 1 failure:
1) mock_should_be_injected_once_and_in_the_best_matching_type(org.mockitousage.bugs.InjectionByTypeShouldFirstLookForExactTypeThenAncestorTest)
java.lang.AssertionError: expected same:<java.lang.Object@5d534f5d> was not:<mockedBean>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotSame(Assert.java:737)
	at org.junit.Assert.assertSame(Assert.java:680)
	at org.junit.Assert.assertSame(Assert.java:691)
	at org.mockitousage.bugs.InjectionByTypeShouldFirstLookForExactTypeThenAncestorTest.mock_should_be_injected_once_and_in_the_best_matching_type(InjectionByTypeShouldFirstLookForExactTypeThenAncestorTest.java:33)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
	at org.mockito.internal.runners.JUnit45AndHigherRunnerImpl.run(JUnit45AndHigherRunnerImpl.java:37)
	at org.mockito.runners.MockitoJUnitRunner.run(MockitoJUnitRunner.java:62)
	at org.junit.runners.Suite.runChild(Suite.java:127)
	at org.junit.runners.Suite.runChild(Suite.java:26)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:160)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:138)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:117)
	at org.junit.runner.JUnitCore.runMain(JUnitCore.java:96)
	at org.junit.runner.JUnitCore.runMainAndExit(JUnitCore.java:47)
	at org.junit.runner.JUnitCore.main(JUnitCore.java:40)

FAILURES!!!
Tests run: 1012,  Failures: 1


real	0m3.897s
user	0m9.376s
sys	0m0.953s
